xio-core/src/main/java/com/xjeffrose/xio/core/SocketAddressHelper.java

```diff
@@ -0,0 +1,34 @@
+package com.xjeffrose.xio.core;
+
+import io.netty.channel.Channel;
+import io.netty.channel.socket.ServerSocketChannel;
+import io.netty.channel.socket.SocketChannel;
+import java.net.InetSocketAddress;
+import javax.annotation.Nullable;
+import lombok.val;
+
+public class SocketAddressHelper {
+
+  @Nullable
+  public static String extractRemoteAddress(Channel channel) {
+    if (channel instanceof ServerSocketChannel) {
+      val inetSocketAddress = ((ServerSocketChannel) channel).remoteAddress();
+      return computeAddress(inetSocketAddress);
+    } else if (channel instanceof SocketChannel) {
+      val inetSocketAddress = ((SocketChannel) channel).remoteAddress();
+      return computeAddress(inetSocketAddress);
+    } else {
+      return null;
+    }
+  }
+
+  private static String computeAddress(InetSocketAddress inetSocketAddress) {
+    if (inetSocketAddress != null) {
+      val inetAddress = inetSocketAddress.getAddress();
+      if (inetAddress != null) {
+        return inetAddress.getHostAddress();
+      }
+    }
+    return null;
```

Comment:
This can just be 
```java
if (inetSocketAddress != null) {
   return inetSocketAddress.getHostString();
}
```

https://docs.oracle.com/javase/8/docs/api/java/net/InetSocketAddress.html#getHostString--

long story short: java 7 made this less tedious.

Reply:
after speaking with pdex, will hold on this change because of rabbit holes

Reply:
done